native/src/routes/jpal-tracking/JpalTracking.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+
+import React, { useEffect, useMemo, useState } from 'react'
+
+import type { ThemeType } from 'build-configs/ThemeType'
+import { Switch, Text, TextInput, View } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import type { StateType } from '../../modules/app/StateType'
+import AppSettings, { defaultSettings } from '../../modules/settings/AppSettings'
+import type { NavigationPropType, RoutePropType } from '../../modules/app/constants/NavigationTypes'
+import type { JpalTrackingRouteType } from 'api-client'
+import styled from 'styled-components/native'
+import LayoutContainer from '../../modules/layout/containers/LayoutContainer'
+import withTheme from '../../modules/theme/hocs/withTheme'
+import Caption from '../../modules/common/components/Caption'
+
+const ThemedText = styled.Text`
+  display: flex;
+  text-align: left;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  padding: 10px 0;
+`
+
+const ErrorText = styled.Text`
+  color: red;
+  font-weight: bold;
+  padding: 10px 0;
+`
+
+const DescriptionContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 15px 0 5px;
+`
+
+const Input = styled(TextInput)`
+  padding: 15px;
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const errorDisplayTime = 5000
+
+const JpalTracking = (props: PropsType, state: StateType) => {
+  const appSettings = useMemo(() => new AppSettings(), [])
+  const [settings, setSettings] = useState(defaultSettings)
+  const [settingsLoaded, setSettingsLoaded] = useState(false)
+  const [settingsUpdated, setSettingsUpdated] = useState(false)
+  const [displayedTrackingCode, setDisplayedTrackingCode] = useState('')
+  const [displayedTrackingEnabled, setDisplayedTrackingEnabled] = useState(false)
+  const [error, setError] = useState(false)
+  const routeTrackingCode = props.route.params.trackingCode
+
+  const { t } = useTranslation(['settings', 'error'])
+
+  useEffect(() => {
+    if (routeTrackingCode) {
+      appSettings.setJpalTrackingCode(routeTrackingCode)
```

Comment:
Shouldn't we display the preset tracking code then? What happens if the route params contains a code different from one already saved in the settings?

Reply:
It gets changed. Thought this would make sense. If one enables tracking and enters a code, and afterwards gets an letter from jpal an opens a deeplink with tracking code, i would expect it to update.

Reply:
It gets changed through loading it from the global app settings afterwards, right? So it is a little race condition, correct?

Yeah, probably you are right as there is no use case where a user clicks a jpal link with tracking code and does not want to change the tracking code. Other opinions @maxammann @Taggotty @svenseeberg? Do you think it is a problem to overwrite a previously set tracking code if a user clicks on a link containing a different tracking code?

Reply:
removed the race condition.